tests/integration/test_distributed_ddl/test.py

```diff
@@ -291,20 +291,32 @@ def test_allowed_databases(test_cluster):
     instance.query("CREATE DATABASE IF NOT EXISTS db2 ON CLUSTER cluster")
 
     instance.query(
-        "CREATE TABLE db1.t1 ON CLUSTER cluster (i Int8) ENGINE = Memory",
+        "CREATE TABLE IF NOT EXISTS db1.t1 ON CLUSTER cluster (i Int8) ENGINE = Memory"
+    )
+    instance.query(
+        "CREATE TABLE IF NOT EXISTS db2.t2 ON CLUSTER cluster (i Int8) ENGINE = Memory"
+    )
+    instance.query(
+        "CREATE TABLE IF NOT EXISTS t3 ON CLUSTER cluster (i Int8) ENGINE = Memory"
+    )
+
+    instance.query(
+        "SELECT * FROM db1.t1",
         settings={"user": "restricted_user"},
     )
 
     with pytest.raises(Exception):
         instance.query(
-            "CREATE TABLE db2.t2 ON CLUSTER cluster (i Int8) ENGINE = Memory",
+            "SELECT * FROM db2.t2",
             settings={"user": "restricted_user"},
         )
+
     with pytest.raises(Exception):
         instance.query(
-            "CREATE TABLE t3 ON CLUSTER cluster (i Int8) ENGINE = Memory",
+            "SELECT * FROM t3",
```

Comment:
why was this test changed so much?

Reply:
The same reason as the [above](https://github.com/ClickHouse/ClickHouse/pull/60117#discussion_r1567522464), so modified the restricted_user.xml this time, and edited the query back.